Files/Extensions/ImageSourceExtensions.cs

```diff
@@ -12,10 +14,43 @@ internal static async Task<byte[]> ToByteArrayAsync(this IInputStream stream)
             {
                 return null;
             }
-            var readStream = stream.AsStreamForRead();
-            var byteArray = new byte[readStream.Length];
-            await readStream.ReadAsync(byteArray, 0, byteArray.Length);
-            return byteArray;
+
+            using var readStream = stream.AsStreamForRead();
+            return await readStream.ToByteArrayAsync();
+        }
+
+        private static async Task<byte[]> ToByteArrayAsync(this Stream stream, CancellationToken cancellationToken = default)
+        {
+            if (stream.CanSeek)
+            {
+                var length = stream.Length - stream.Position;
+                if (length == 0)
+                    return Array.Empty<byte>();
+
+                var buffer = new byte[length];
+                var actualLength = await TryReadAllAsync(stream, buffer, cancellationToken).ConfigureAwait(false);
+                Array.Resize(ref buffer, actualLength);
+                return buffer;
+            }
+
+            using var memoryStream = new MemoryStream();
+            await stream.CopyToAsync(memoryStream, bufferSize: 81920, cancellationToken).ConfigureAwait(false);
```

Comment:
Curious: why do you distinguish between a seekable and non-seekable stream? Isn't the second part ok for both cases?

Reply:
It's for perf reasons. The MemoryStream may need to grow its local array multiple times (and copy data from the old array to the new one) when copying the original stream. If you know the length of the stream in advance, you can allocate the array once and fill it. => So, you reduce the number of allocations

Reply:
What about using `memoryStream = new MemoryStream(LENGTH)`?

Reply:
I can do it using the MemoryStream if you think it's easier to maintain. In terms of perf, it will be a little bit slower (1 more allocation and a more complex logic in the `MemoryStream.Write` method), but the difference should be negligeable in this case

Reply:
Thanks for the explanation, I'd go with the simple method (new MemoryStream(stream.Length)) here.

Reply:
The latest commit should be ok 🙂